packages/slate/src/utils/find-closest-difference.js

```diff
@@ -0,0 +1,39 @@
+/**
+ * Find the smallest node covering all difference between previousNode and node
+ * @param {Node|Text} previousMNode
+ * @param {Node|Text} node
+ * @return {null|Node|Text}
+ *
+ */
+
+function findClosestDifference(previousNode, node) {
+  if (node === previousNode) return null
+  if (node.object !== previousNode.object) return node
+  if (node.object === 'text') return node
+  if (node.nodes.size !== previousNode.nodes.size) return node
+  if (node.key !== previousNode.key) return node
+  if (node.data !== previousNode.data) return node
+
+  const first = node.nodes.findIndex(
+    (child, index) => child !== previousNode.nodes.get(index)
+  )
+
+  const last = node.nodes.findLastIndex(
+    (child, index) => child !== previousNode.nodes.get(index)
+  )
+
+  if (first !== last) return node
+
+  return findClosestDifference(
+    previousNode.nodes.get(first),
+    node.nodes.get(first)
+  )
+}
```

Comment:
About the two child thing, is it handled already by 
```
if (node.nodes.size !== previousNode.nodes.size) return node

```
line 13?

Reply:
No, i meant if the number of children of the block stayed the same but two or more of them changed

Reply:
@CameronAckermanSEL I am curious about that. Could you example that corner case?

Reply:
It's not a corner case you haven't handled, the conditional on line 25 addresses it (because if the first changed node is different from the last changed node, then this parent needs to be normalized), so it's good to go. The point was that it took a while to get to that conclusion, so a comment could be used to indicate what that check captures.